### DISCUSSION TITLE
Unexpected empty environment variables

### DISCUSSION THREAD

**Discussion by EmilyGraceSeville7cf** [community/community#24610] [Apps, API and Webhooks]
<p>Hello! Why <code>GITHUB_API_URL</code> and <code>GITHUB_REPOSITORY_OWNER</code> are shown as empty in this <a href="https://github.com/EmilySeville7cfg/personal-automation-helpers/runs/6828071824?check_suite_focus=true#step:3:17" rel="noopener nofollow ugc">workflow run</a>? How to fix this problem? I don’t want hardcode any strings and wanna use environment variables.</p>

**Comment by airtower-luna** [comment]
<p>Because you’re calling a reusable workflow for that job. The <a href="https://docs.github.com/en/actions/using-workflows/reusing-workflows#limitations">“Limitations” section in the documentation on those</a> says:</p>
<blockquote>
<p>Any environment variables set in an <code>env</code> context defined at the workflow level in the caller workflow are not propagated to the called workflow.</p>
</blockquote>
<p>From a glance at the code I’m not sure why you’re using a reusable workflow there, maybe a <a href="https://docs.github.com/en/actions/creating-actions/creating-a-composite-action">composite action</a> would do the job better? That would have access to the environment variables of a job that calls it. :smiley_cat:</p>
<p>If you really need a reusable workflow, you’ll have to pass the data you need using <a href="https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow">inputs for the reusable workflow</a>.</p>

**Comment by EmilyGraceSeville7cf** [comment]
<blockquote>
<p>Any environment variables set in an <code>env</code> context defined at the workflow level in the caller workflow are not propagated to the called workflow.</p>
</blockquote>
<p><a class="mention" href="/airtower-luna">@airtower-luna</a>, but they are defied in <a href="https://github.com/EmilySeville7cfg/personal-automation-helpers/blob/feature/5/.github/workflows/check-markdown.yaml#L20" rel="noopener nofollow ugc">called</a> workflow. :thinking: There are no variables <a href="https://github.com/EmilySeville7cfg/personal-automation-helpers/blob/feature/5/.github/workflows/ci.yaml#L7" rel="noopener nofollow ugc">defined</a> in the caller workflow.</p>
<blockquote>
<p>If you reuse a workflow from a different repository, any actions in the called workflow run as if they were part of the caller workflow.</p>
</blockquote>
<p>Is it right that even <code>env</code> is in my called workflow it behave like if it was defined in the caller workflow?</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="EmilySeville7cfg" data-post="3" data-topic="255767">
<div class="title">
<div class="quote-controls"></div>
 EmilySeville7cfg:</div>
<blockquote>
<p>but they are defied in <a href="https://github.com/EmilySeville7cfg/personal-automation-helpers/blob/feature/5/.github/workflows/check-markdown.yaml#L20">called</a> workflow.</p>
</blockquote>
</aside>
<p>The definitions in the <code>env</code> block for that step try to access environment variables and even secrets that are automatically set <em>in the caller workflow</em> (all the <code>env.</code> and <code>secret.</code> things you use there). The documentation I pointed out explicitly says those are not passed on to the called workflow.</p>

**Comment by djbrown** [comment]
You could use [GitHub Actions Variables](https://docs.github.com/en/actions/learn-github-actions/variables) (currently still Beta though)

**Reply by EmilyGraceSeville7cf** [reply]
Thanks! 😄

**Reply by djbrown** [reply]
@EmilySeville7cfg you're welcome ;)
there is currently a "problem" with accessing variables in workflows triggered by public fork pull requests:
https://github.com/orgs/community/discussions/44322
i hope GitHub resolves it soon
